### PR TITLE
compiler.h: add MSVC(Microsoft Visual C++)-specific definitions

### PR DESCRIPTION
**Pull request by anchao**
## Summary

compiler.h: add MSVC(Microsoft Visual C++)-specific definitions

Signed-off-by: chao an <anchao@xiaomi.com>

## Impact

N/A

## Testing

cmake + sim/windows

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/nuttx/compiler.h [hunk 1]
```diff
@@ -745,6 +745,69 @@
...
+#  define CODE
+#  define IOBJ
+#  define IPTR
+
+#  undef  CONFIG_SMALL_MEMORY
+#  undef  CONFIG_LONG_IS_NOT_INT
+#  undef  CONFIG_PTR_IS_NOT_INT
+#  define CONFIG_HAVE_FLOAT 1
+#  undef  CONFIG_HAVE_DOUBLE
+#  undef  CONFIG_HAVE_LONG_DOUBLE
```
Comment (pkarashchenko) [inline include/nuttx/compiler.h:803]: So MSC does not have long double?

Comment (anchao) [reply include/nuttx/compiler.h:803]: long double should be supported, I just enabled the flags which I interested in.

Comment (pkarashchenko) [reply include/nuttx/compiler.h:803]: But those defines are more what compiler supports or not. Otherwise those would be located in Kconfig

Comment (anchao) [reply include/nuttx/compiler.h:803]: OK, let me enable long double defines.

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
Let's ignore the false alarm.

**Pull request merged**